src/components/RouteRow.tsx

```diff
@@ -0,0 +1,37 @@
+import { Box, Flex, Text } from 'native-base';
+import { Route } from '../xplat/types/route';
+import { useEffect, useState } from 'react';
+import { createSearchParams, useNavigate } from 'react-router-dom';
+import { Pressable } from 'react-native';
+
+type Props = {
+    route: Route;
+};
+export const RouteRow = ({ route }: Props) => {
+    const [name, setName] = useState<string>('');
+    const [grade, setGrade] = useState<string>('');
+
+    useEffect(() => {
+        Promise.all([route.getName().then(setName), route.getGradeDisplayString().then(setGrade)]);
```

Comment:
This doesn't really have an effect unless it's awaited. Although correct, this would more idiomatically be written sequentially.
```suggestion
        route.getName().then(setName);
        route.getGradeDisplayString().then(setGrade);
```